.github/workflows/pull-request.yml

```
@@ -348,3 +348,55 @@
+            RESPONSE=$(docker exec talawa-api-1 curl -s -X POST http://127.0.0.1:4000/graphql -H "Content-Type: application/json" -d '{"query":"{__typename}"}' 2>/dev/null || echo "Connection failed")
+            if echo "$RESPONSE" | grep -q '__typename'; then
+              echo "GraphQL endpoint is available!"
+              exit 0
+            fi
+            echo "GraphQL endpoint not ready. Retrying in 2 seconds..."
+            sleep 2
+          done
+          echo "GraphQL endpoint did not become available within the expected time."
+          exit 1
+      - name: Cleanup - Free ports by stopping containers
+        run: docker compose down
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)